### ISSUE TITLE
Selecting Windows does not bring them to the front

### ISSUE THREAD

**Issue by Duion** [GarageGames/Torque3D#1823]
When you select GUI windows, they are supposed to be moved to the front as seen here in the source code:
https://github.com/GarageGames/Torque3D/blob/development/Engine/source/gui/containers/guiWindowCtrl.cpp#L736
But this does not work currently.
To reproduce this issue you need two GUI windows on the screen and both with "GuiModelessDialogProfile" since this does not block the other GUIs. I used:
Canvas.pushDialog( OptionsDlg ); and Canvas.pushDialog(PostFXManager); in console, you have to set the OptionsDlg to "GuiModelessDialogProfile" before.
Then you bring one window in front of the other and select the window behind and you will see that it will not be brought to the front, which it is supposed to do.

**Comment by Areloch** [comment]
It's based on the parent of the window.

In the example you gave, the dialogs have different parent guiControls for each window, so they're separate parent objects, and thus don't reorder. If you have the guiWindowCtrl under the same parent object, then they re-order fine.

**Comment by Duion** [comment]
You are right, but you have to click the top bar on the window, then they reorder.
Now do you have any idea how to bring this feature to work with separate parent objects?
